src/helpers/Html.php

```diff
@@ -760,11 +760,27 @@ function(array $match) use ($namespace, $ids): string {
     private static function _escapeTextareas(string &$html): array
     {
         $markers = [];
-        $html = preg_replace_callback('/(<textarea\b[^>]*>)(.*?)(<\/textarea>)/is', function(array $matches) use (&$markers) {
+        $data = preg_replace_callback('/(<textarea\b[^>]*>)(.*?)(<\/textarea>)/is', function(array $matches) use (&$markers) {
             $marker = '{marker:' . StringHelper::randomString() . '}';
             $markers[$marker] = $matches[2];
             return $matches[1] . $marker . $matches[3];
         }, $html);
+
+        // this can be null if the pcre.backtrack_limit is reached
+        // https://github.com/craftcms/cms/issues/13083
+        if ($data !== null) {
+            /* @phpstan-var $data string */
+            $html = $data;
+        } else {
+            preg_match('/(.*?)(<textarea\b[^>]*>)/is', $html, $matches);
+            $start = $matches[0];
+            preg_match('/(<\/textarea>.*)/is', $html, $matches);
+            $end = $matches[0];
+            $marker = '{marker:' . StringHelper::randomString() . '}';
+            $markers[$marker] = str_replace([$start, $end], '', $html);
+            $html = $start . $marker . $end;
```

Comment:
This looks like it’s assuming there is only 1 `<textarea>` tag. Think you’d need to keep doing this in a loop until `preg_match` returns `0`.

And if this method is more performant than `preg_replace_callback()`, we should probably just go straight to using it, rather than just as a fallback if `$data` is` null`.

Reply:
I adjusted it for multiple `<textarea>` tags and dropped the `preg_replace_callback()` entirely.

Tested against "more then 1 million chars" as metioned in the issue and also against 16MB of text in a field and this version is still able to process it all.